sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/DefaultChannelTest.java

```diff
@@ -1104,6 +1080,34 @@ public void checkPendingLogsStoresStartTime() {
         verify(mAppCenterHandler).postDelayed(any(Runnable.class), eq(CUSTOM_INTERVAL));
     }
 
+    @Test
+    public void checkPendingLogsStoresStartTimeWhenPaused() {
+
+        /* Mock current time. */
+        long now = 1;
+        when(System.currentTimeMillis()).thenReturn(now);
+
+        /* Create channel and group. */
+        Persistence mockPersistence = mock(Persistence.class);
+        when(mockPersistence.countLogs(TEST_GROUP)).thenReturn(0);
+        AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+        DefaultChannel channel = new DefaultChannel(mock(Context.class), UUIDUtils.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+        channel.addGroup(TEST_GROUP, 10, CUSTOM_INTERVAL, MAX_PARALLEL_BATCHES, mockIngestion, mock(Channel.GroupListener.class));
+        verifyStatic(never());
+        SharedPreferencesManager.putLong(eq(START_TIMER_PREFIX + TEST_GROUP), anyLong());
+        verify(mockIngestion, never()).sendAsync(anyString(), anyString(), any(UUID.class), any(LogContainer.class), any(ServiceCallback.class));
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
+
+        channel.pauseGroup(TEST_GROUP, null);
+        channel.enqueue(mock(Log.class), TEST_GROUP, Flags.DEFAULTS);
+
+        /* Verify that timer starts and current time is saved into preferences. */
+        verifyStatic();
+        SharedPreferencesManager.putLong(eq(START_TIMER_PREFIX + TEST_GROUP), eq(now));
+        verify(mockIngestion, never()).sendAsync(anyString(), anyString(), any(UUID.class), any(LogContainer.class), any(ServiceCallback.class));
+        verify(mAppCenterHandler, never()).postDelayed(any(Runnable.class), anyLong());
```

Comment:
This test is incomplete, we must simulate some time has passed then resume and check sent immediately.

Reply:
Also please move this new test to `DefaultChannelPauseAndResumeTest`, we already split the test some time back as the main class is cluttered.

Let's try to keep them split by themes.